st2actions/st2actions/container/base.py

```diff
@@ -60,12 +60,21 @@ def dispatch(self, liveaction_db):
         LOG.debug('Runner instance for RunnerType "%s" is: %s', runnertype_db.name, runner)
 
         # Process the request.
-        if liveaction_db.status == action_constants.LIVEACTION_STATUS_CANCELING:
-            liveaction_db = self._do_cancel(runner=runner, runnertype_db=runnertype_db,
-                                            action_db=action_db, liveaction_db=liveaction_db)
-        else:
-            liveaction_db = self._do_run(runner=runner, runnertype_db=runnertype_db,
-                                         action_db=action_db, liveaction_db=liveaction_db)
+        funcs = {
+            action_constants.LIVEACTION_STATUS_REQUESTED: self._do_run,
+            action_constants.LIVEACTION_STATUS_SCHEDULED: self._do_run,
+            action_constants.LIVEACTION_STATUS_RUNNING: self._do_run,
+            action_constants.LIVEACTION_STATUS_CANCELING: self._do_cancel,
+            action_constants.LIVEACTION_STATUS_PAUSING: self._do_pause,
+            action_constants.LIVEACTION_STATUS_RESUMING: self._do_resume
+        }
+
+        liveaction_db = funcs[liveaction_db.status](
```

Comment:
Maybe try / catch which throws a more friendly exception in case of invalid status?

I know this shouldn't really happen since it would indicate a programmer error, but still.